modApiServer/src/org/aion/api/server/rpc3/AionChainHolder.java

```diff
@@ -96,19 +102,63 @@ public boolean submitSignature(byte[] signature, byte[] sealHash) {
         if (!isUnityForkEnabled()) throw new UnsupportedOperationException();
         else {
             StakingBlock stakingBlock = chain.getBlockchain().getCachingStakingBlockTemplate(sealHash);
-            stakingBlock.seal(signature, stakingBlock.getHeader().getSigningPublicKey());
 
-            //AKI-648 reject the block add into the kernel if the timestamp of the block is in the future.
-            boolean isValidTimestamp = futureBlockRule.validate(stakingBlock.getHeader(), new ArrayList<>());
+            logger.debug(
+                    "submitSignature: sig[{}], sealHash[{}], block[{}]",
+                    ByteUtil.toHexString(signature),
+                    ByteUtil.toHexString(sealHash),
+                    stakingBlock);
 
-            final boolean sealed = isValidTimestamp && addNewBlock(stakingBlock);
+            if (!stakingBlock.isSealed()
+                    && Arrays.equals(sealHash, stakingBlock.getHeader().getMineHash())) {
+                stakingBlock.seal(signature, stakingBlock.getHeader().getSigningPublicKey());
 
-            if (sealed) {
+                addSealedBlockToPool(stakingBlock);
                 logSealedBlock(stakingBlock);
-            }else {
+                return true;
+            } else {
                 logFailedSealedBlock(stakingBlock);
+                return false;
             }
-            return sealed;
+        }
+    }
+
+    private void addSealedBlockToPool(StakingBlock stakingBlock) {
+        try {
+            blockSubmitThread.submit(
+                    () -> {
+                        long currentTimeMillis = System.currentTimeMillis();
+                        long blockTimeMillis =
+                                TimeUnit.SECONDS.toMillis(stakingBlock.getTimestamp());
+                        long diff = blockTimeMillis - currentTimeMillis;
+
+                        if (diff > TimeUnit.MINUTES.toMillis(10)) {
+                            logger.info(
+                                    "discard the sealed block due to long timestamp(ms): blockTimeStamp[{}], exceed the current time[{}]",
+                                    blockTimeMillis,
+                                    diff);
+                            return;
+                        } else if (diff > 0) {
+                            logger.info("the block {} runnable sleep(ms): {}", stakingBlock.getNumber(), diff);
+                            try {
+                                Thread.sleep(diff);
```

Comment:
What is the purpose of this sleep? Since we're using a thread pool, couldn't we just give the threads actionable task instead of blocking them with a sleep method?